api_tests/preprints/views/test_preprint_detail.py

```diff
@@ -230,449 +263,511 @@ def test_noncontrib_cannot_set_primary_file(self):
         unpublished.reload()
-        assert_true(unpublished.is_published)
+        assert unpublished.is_published
 
     @mock.patch('website.preprints.tasks.on_preprint_updated.s')
-    def test_update_preprint_task_called_on_api_update(self, mock_on_preprint_updated):
-        update_doi_payload = build_preprint_update_payload(self.preprint._id, attributes={'doi': '10.1234/ASDFASDF'})
+    def test_update_preprint_task_called_on_api_update(self, mock_on_preprint_updated, app, user, preprint, url):
+        update_doi_payload = build_preprint_update_payload(preprint._id, attributes={'doi': '10.1234/ASDFASDF'})
 
-        self.app.patch_json_api(self.url, update_doi_payload, auth=self.user.auth)
-        self.preprint.node.reload()
+        app.patch_json_api(url, update_doi_payload, auth=user.auth)
+        preprint.node.reload()
 
         assert mock_on_preprint_updated.called
 
-class TestPreprintUpdateLicense(ApiTestCase):
-
-    def setUp(self):
-        super(TestPreprintUpdateLicense, self).setUp()
-
-        self.admin_contributor = AuthUserFactory()
-        self.rw_contributor = AuthUserFactory()
-        self.read_contributor = AuthUserFactory()
-        self.non_contributor = AuthUserFactory()
-
-        self.preprint_provider = PreprintProviderFactory()
-        self.preprint = PreprintFactory(creator=self.admin_contributor, provider=self.preprint_provider)
-
-        self.preprint.node.add_contributor(self.rw_contributor, auth=Auth(self.admin_contributor))
-        self.preprint.node.add_contributor(self.read_contributor, auth=Auth(self.admin_contributor), permissions=['read'])
-        self.preprint.node.save()
-
-        self.cc0_license = NodeLicense.find_one(Q('name', 'eq', 'CC0 1.0 Universal'))
-        self.mit_license = NodeLicense.find_one(Q('name', 'eq', 'MIT License'))
-        self.no_license = NodeLicense.find_one(Q('name', 'eq', 'No license'))
-
-        self.preprint_provider.licenses_acceptable = [self.cc0_license, self.no_license]
-        self.preprint_provider.save()
-
-        self.url = '/{}preprints/{}/'.format(API_BASE, self.preprint._id)
-
-    def make_payload(self, node_id, license_id=None, license_year=None, copyright_holders=None):
-        attributes = {}
-
-        if license_year and copyright_holders:
-            attributes = {
-                'license_record': {
-                    'year': license_year,
-                    'copyright_holders': copyright_holders
+@pytest.mark.django_db
+class TestPreprintUpdateLicense:
+
+    @pytest.fixture()
+    def admin_contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def write_contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def read_contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def non_contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def cc0_license(self):
+        return NodeLicense.find_one(Q('name', 'eq', 'CC0 1.0 Universal'))
```

Comment:
These should be changed to use Django's Q instead of modm Q's. Bug me about that if you don't know how to convert.